### PR TITLE
feat(docs): add documentation to upgrade from Stark 10 (Angular 7) to Stark 12 (Angular 16)

### PR DESCRIPTION
**Pull request by mhenkens**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[x] Other... Please describe: Documentation for migration
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: docs/MIGRATION-ESLINT.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+### package.json
+
+```json
+{
+  ...
+  "devDependencies": {
+    "@nationalbankbelgium/eslint-config": "^12.0.0",
+    ...
+    "eslint-config-prettier": "^8.8.0",
```
Comment (SuperITMan) [inline docs/MIGRATION-ESLINT.md:27]: Could we recommend to directly use the current latest version `9.1.0` ?

**Review by SuperITMan — CHANGES_REQUESTED** [review]
Small changes to apply + some questions

File: docs/MIGRATION_ANGULAR_16.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+}
+```
+
+### update eslint-config in package.json
+
+```json
+...
+"devDependencies": {
+	...
+	"@nationalbankbelgium/eslint" : "^16.0.0",
```
[the remark this answers is not in the document]
Comment (mhenkens) [reply docs/MIGRATION_ANGULAR_16.md:84]: I've tried but a received error when npm install before run schematics. It's better to execute in 2 step

File: docs/MIGRATION_ANGULAR_15.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+}
+```
+
+### update eslint-config in package.json
+
+```json
+...
+"devDependencies": {
+	...
+	"@nationalbankbelgium/eslint" : "^15.0.1",
```
[the remark this answers is not in the document]
Comment (mhenkens) [reply docs/MIGRATION_ANGULAR_15.md:82]: I've tried but a received error when npm install before run schematics. It's better to execute in 2 step

File: docs/MIGRATION_ANGULAR_14.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+}
+```
+
+### update eslint-config in package.json
+
+```json
+...
+"devDependencies": {
+	...
+	"@nationalbankbelgium/eslint" : "^14.0.0-rc.0",
```
[the remark this answers is not in the document]
Comment (mhenkens) [reply docs/MIGRATION_ANGULAR_14.md:82]: I've tried but a received error when npm install before run schematics. It's better to execute in 2 step

File: docs/MIGRATION_ANGULAR_13.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+}
+```
+
+### update eslint-config in package.json
+
+```json
+...
+"devDependencies": {
+	...
+	"@nationalbankbelgium/eslint" : "13.0.0",
```
[the remark this answers is not in the document]
Comment (mhenkens) [reply docs/MIGRATION_ANGULAR_13.md:82]: I've tried but a received error when npm install before run schematics. It's better to execute in 2 step

File: docs/MIGRATION-ESLINT.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+	"lint-es:fix": "ng lint --fix",
+
+	<!-- to be removed -->
+	"lint-ts" : ...
+```
+
+### Set the lodash version
+
+```json
+	"@types/lodash":"4.14.195",
```
[the remark this answers is not in the document]
Comment (mhenkens) [reply docs/MIGRATION-ESLINT.md:91]: I added the instruction in the angular_16 guide

File: docs/MIGRATION-ESLINT.md (hunk 1 above)
Comment (mhenkens) [reply docs/MIGRATION-ESLINT.md:27]: I've tested in angular 16 but not in previous versions so i added in angular 16

File: docs/MIGRATION-ESLINT.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
+# Migration from TSLINT to ESLINT
+
+## install eslint schematics
+
+please use `gitbash` to execute commands.
```
[the remark this answers is not in the document]
Comment (SuperITMan) [reply docs/MIGRATION-ESLINT.md:5]: It seems you missed to commit your change about this.
Could you adapt the following line as in the other files ?

```
please use `gitbash` to execute commands.
```

**Review by SuperITMan — CHANGES_REQUESTED** [review]
Small changes to apply. 

Could you also run prettier on all your new md files in order to fix GitHub Actions ? 

Finally, could you reword your commit message from 
```txt
feat(docs): add doccumentation to upgrade from angualar 8 to 16
```
to
```txt
feat(docs): add documentation to upgrade from Stark 10 (Angular 7) to Stark 12 (Angular 16)
```

File: docs/MIGRATION_ANGULAR_16.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
```
[the remark this answers is not in the document]
Comment (mhenkens) [reply docs/MIGRATION_ANGULAR_16.md:84]: we are still on "^8.8.0"

**Pull request merged**